web-client/src/ducks/timeline/functions.ts

```diff
@@ -1,15 +1,114 @@
 import { firestore } from 'src/firebase';
-import { TimelineItemFirestoreConverter } from 'src/models/requests/timeline';
+// eslint-disable-next-line no-unused-vars
+import { IOffer, Offer } from 'src/models/offers';
+import { IRequest, Request, RequestFirestoreConverter, RequestRefType } from 'src/models/requests';
+import { IRequestWithOffersAndTimelineItems, RequestWithOffersAndTimeline } from 'src/models/requests/RequestWithOffersAndTimeline';
+// eslint-disable-next-line no-unused-vars
+import { ITimelineItem, TimelineItemAction, TimelineItemFirestoreConverter } from 'src/models/requests/timeline';
 
-import { IgetTimeline } from './types';
+const getRequest = async (requestRef: RequestRefType): Promise<IRequest> => {
+  console.log('abcd');
+  let requestData;
+    await firestore
+      .collection('requests')
+      .doc('1EQlpGu0QQMBji6MEPZ7')
+      .withConverter(RequestFirestoreConverter)
+      .get()
+      .then(snapshot => {
+          console.log('xxx',snapshot);
+          requestData = {
+          id: snapshot.id,
+          _requestRef: snapshot.ref,
+          // TODO:(es) Convert to request here?
+          ...snapshot.data(),
+        };
+      });
 
-export const observeTimelinesFunc = (
-  nextValue: Function,
-  payload: IgetTimeline,
-): firebase.Unsubscribe =>
+  console.log('xxxx', requestData, 'xxxz', requestRef);
+  const request = Request.factoryFromUnderscore({ ...requestData, requestRef });
+  console.log('xxxz2', request);
+  return request as IRequest;
+};
+
+const getOffersForRequest = async (requestRef: RequestRefType) => {
+  console.log('zz6', requestRef);
+  const result = await requestRef
+    .collection('offers')
+    .get();
+  console.log('zz6.1');
+  const offers: Record<string, IOffer> = {};
+
+  for (const doc of result.docs) {
+    const offer = Offer.factory(doc.data() as IOffer);
```

Comment:
The CAV who is responding does not need to know distance to privileged info, as we discussed, so this logic would change.